### ISSUE TITLE
OpenBST feature parity tracking

### ISSUE THREAD

**Issue by utybo** [utybo/StoryFX#4]
This issue tracks the feature parity between StoryFX's format (story.kts) and its predecessor, OpenBST's .bst format

*All* of the features from OpenBST should at least have some counterpart in StoryFX.

- [x] Nodes
  Done and fully implemented.
- [x] Options with simple NND
  Done and fully implemented
- [x] if-NND
  Indirect benefit from Kotlin, just put an if in the nodeRef call
- [x] Markdown support
  Done and fully implemented
- [ ] HTML support
  Unsure if useful, can simply use Markdown with just HTML in it.
- [x] Author and title properties
  Done and partially implemented (authors are not exposed yet)
- [ ] NSFW property
  TODO
- [x] Font support
  Done for preset fonts (two available: Mulli and Merriweather
- [ ] Custom color support
  Unsure if useful, can simply use HTML and/or custom CSS
- [ ] Supertools (reload, reset)
  TODO
- [x] Custom initial node
  Done and fully implemented
- [x] Scripting (actions)
  Done and fully implemented. Indirect benefits from Kotlin.
- [x] Scripting (checkers)
  Done and fully implemented. Indirect benefits from Kotlin.
- [x] Scripting (input)
  Done and fully implemented.
- [x] Logical nodes
  Won't do, you can simply use Kotlin functions or just full functions in lambdas if reusability does not matter
- [ ] Virtual nodes
  Won't do, can be done fairly easily in Kotlin, but can probably be improved
- [x] Aliasing
  Unsure if useful since nodes have string ids
- [x] Auto-id (string ID)
  Done and fully implemented. All nodes actually use string ids.
- [ ] User Info Bar. 
  Unlikely to be as thorough as it was in OpenBST, and unlikely to be implemented at all due to its complexity
- [x] Resource manager (BRM)
  Done, still needs work
- [x] Custom background (IMG)
  Done and fully implemented
- [ ] Sounds (SSB)
  TODO
- [ ] Data files (BDF)
  TODO, unsure of format
- [ ] JavaScript Engine (JSE)
  Unsure if useful since we are using Kotlin
- [ ] Cross-file multi-story integration (XBF)
  TODO
- [ ] HTML Bridge and security related actions (HTB)
  TODO
- [ ] JavaScript file support (XSF)
  Unsure if useful, but loading other story.kts files is TODO
- [ ] Save states
  Will be fairly complicated
- [ ] Jump to Node
  TODO
- [ ] Variable Watcher
  TODO, will be limited
- [ ] Toggle background visibility
  Unsure if useful
- [ ] Popup view for seeing the background up close and personal
  TODO, unsure of how to do it
- [ ] Mute
  TODO once sound system is in place
- [x] Close the tab
  Done, side effect of using JavaFX which makes tabs closeable by default
- [ ] BSP package support
  TODO, probably with a new format
- [ ] Custom CSS file
  TODO
- [ ] Internal IMG backgrounds
  Unlikely to happen since that just adds bloat to the file size
